src/processor/reconnection-graph.service.ts

```diff
@@ -125,4 +140,147 @@ export class ReconnectionGraphService implements OnApplicationBootstrap, OnAppli
+        importBundleBuilder
+          .withDsnpUserId(dsnpUserId.toString())
+          .withSchemaId(private_follow_schema_id)
+          .withDsnpKeys(dsnpKeys)
+          .withPageData(privateFollow.page_id.toNumber(), privateFollow.payload, privateFollow.content_hash.toNumber())
+          .withGraphKeyPair(GraphKeyType.X25519, graphKeyPair.publicKey, graphKeyPair.privateKey)
+          .build()
+      )
+    );
+  
+    importBundles.push(
+      ...privateFriendships.map((privateFriendship) =>
+        importBundleBuilder
+          .withDsnpUserId(dsnpUserId.toString())
+          .withSchemaId(private_friendship_schema_id)
+          .withDsnpKeys(dsnpKeys)
+          .withPageData(privateFriendship.page_id.toNumber(), privateFriendship.payload, privateFriendship.content_hash.toNumber())
+          .withGraphKeyPair(GraphKeyType.X25519, graphKeyPair.publicKey, graphKeyPair.privateKey)
+          .build()
+      )
+    );
+  
+    return importBundles;
+  }
+  
+
+  async formConnections(dsnpUserId: MessageSourceId, providerId: MessageSourceId, graphSdkConfig: Config, graphConnections: ProviderGraph[]): Promise<ConnectAction[]> {
+    const dsnpKeys = await this.formDsnpKeys(dsnpUserId, graphSdkConfig);
+    const public_follow_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Follow, PrivacyType.Public);
+    const public_friendship_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Friendship, PrivacyType.Public);
+    const private_follow_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Follow, PrivacyType.Private);
+    const private_friendship_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Friendship, PrivacyType.Private);
+
+    let actions: ConnectAction[] = [];
+    graphConnections.forEach((connection) => {
+      let schemaId: number;
+      const connectionType = connection.connectionType.toLowerCase();
+      const privacyType = connection.privacyType.toLowerCase();
+      
+      switch (connectionType) {
+        case 'follow':
+          schemaId = privacyType === 'public' ? public_follow_schema_id : private_follow_schema_id;
+          break;
+        case 'friendship':
+          schemaId = privacyType === 'public' ? public_friendship_schema_id : private_friendship_schema_id;
+          break;
+        default:
+          throw new Error(`Unrecognized connection type: ${connectionType}`);
+      }
+
+      switch(connection.direction) {
+        case 'connectionTo':
+          actions.push({
+            ownerDsnpUserId: dsnpUserId.toString(),
+            dsnpKeys: dsnpKeys,
+            connection: {
+              dsnpUserId: connection.dsnpId,
+              schemaId,
+            } as Connection,
+          } as ConnectAction);
+        case 'connectionFrom':{
+          const { key: jobId, data } = createGraphUpdateJob(connection.dsnpId, providerId, false);
```

Comment:
nit: use constant `UpdateTransitiveGraph` or `SkipTransitiveGraph` instead of `true/false` for readability